### PR TITLE
propper svg/math elements

### PR DESCRIPTION
**Pull request by choltreppe**
ok, so here it is. I did the same test as in the .yml file so it should be all good.
edit: I ticked the version number, hope thats correct

### CODE REVIEW

File: karax/karax.nim [hunk 1]
```diff
@@ -195,7 +198,13 @@ proc toDom*(n: VNode; useAttachedNode: bool; kxi: KaraxInstance = nil): Node =
     attach n
     return result
   else:
-    result = document.createElement(toTag[n.kind])
+    result =
+      if n.kind in svgElements:
+        document.createElementNS(svgNamespace, $n.kind)
```
Comment (geotre) [inline karax/karax.nim:203]:
```suggestion
        document.createElementNS(svgNamespace, toTag[n.kind])
```

for consistency if not correctness

File: karax/karax.nim [hunk 2]
```diff
@@ -195,7 +198,13 @@ proc toDom*(n: VNode; useAttachedNode: bool; kxi: KaraxInstance = nil): Node =
...
+      elif n.kind in mathElements:
+        document.createElementNS(mathNamespace, $n.kind)
```
Comment (geotre) [inline karax/karax.nim:205]:
```suggestion
        document.createElementNS(mathNamespace, toTag[n.kind])
```

**Review by geotre — CHANGES_REQUESTED** [review]

File: karax/vdom.nim [hunk 3]
```diff
@@ -68,6 +68,12 @@ const
   selfClosing = {area, base, br, col, embed, hr, img, input,
     link, meta, param, source, track, wbr}
 
+  svgElements* = {animate .. view}
+  mathElements* = {maction .. semantics}
+
+  svgNamespace* = "http://www.w3.org/2000/svg"
+  mathNamespace* = "http://www.w3.org/1998/Math/MathML"
+
```
[the remark this answers is not in the document]
Comment (choltreppe) [reply karax/vdom.nim:76]: ah yep thats smart

File: karax/karax.nim (hunk 1 above)
Comment (geotre) [reply karax/karax.nim:203]: Good catch

My concern with using `$n.kind` is that some svg elements (from a quick look at the list - `stext`, `discard`, `set`) may not work correctly

I'm not sure if toTag needs to convert to uppercase, tests pass without it. Any idea @Araq? It was added in [this PR](https://github.com/karaxnim/karax/commit/be0f9c9e08b85591b38bd784bb3f6013e6856a18)

Comment (choltreppe) [reply karax/karax.nim]: no they all have the correct string value. for example `stext = "text"`. all `toTag` does is uppercasing and casting to `kstring`. I updated it now to not uppercase

**Review by geotre — APPROVED** [review]

**Comment by geotre** [comment]
Looks great. Good job @choltreppe

**Pull request merged**